Expired/Views/CategoriesView.swift

```diff
@@ -43,6 +68,22 @@ struct CategoriesView: View {
             }
         }
     }
+    private func deleteCategory(indexSet: IndexSet){
+        withAnimation {
+            indexSet.map{productStore.categories[$0]}.forEach{ category in
+                if(category.products?.count == 0){
+                    viewContext.delete(category)
+                }else{
+                   showingNotDeleteAlert =  true
+                }
+                productStore.save(viewContext)
+            }
+        }
+    }
```

Comment:
```suggestion

    private func deleteCategory(indexSet: IndexSet) {
        withAnimation {
            indexSet.map{productStore.categories[$0]}.forEach{ category in
                if category.products?.count ?? 0 > 0 {
                    showingNotDeleteAlert = true
                } else {
                   viewContext.delete(category)
                }
            }
            productStore.save(viewContext)
        }
    }

```

- space matters
- Avoid invoking `productStore.save` multiple times by moving it outside of the loop